scripts/west_commands/runners/bossac.py

```diff
@@ -110,8 +100,13 @@ def get_chosen_code_partition_node(self):
             raise RuntimeError(error_msg)
 
         # Load the devicetree.
-        with open(edt_pickle, 'rb') as f:
-            edt = pickle.load(f)
+        try:
+            with open(edt_pickle, 'rb') as f:
+                edt = pickle.load(f)
+        except ModuleNotFoundError:
+            error_msg = "could not load devicetree, something may be wrong " \
```

Comment:
probably use parenthesis and no `+` to wrap string?

personally would just let the exception raise thru as you're dropping exception information here (`ModuleNotFoundError` becomes `RuntimeError`)

Reply:
I added this check as a hedge against the "twister crash in CI in mysterious ways" as noted by Marti.